### PR TITLE
translate-c: fixed casting between pointers and ints

### LINKED ISSUE #17427: error: @bitCast size mismatch: destination type 'usize' has 64 bits but source type 'c_long' has 32 bits
### Zig Version

0.12.0-dev.3+9c05810be

### Steps to Reproduce and Observed Behavior

1. `git clone https://github.com/kora-org/headstart`
2. `zig build -Darch=riscv64`
3. This error should pop up:
```
zig build-exe headstart Debug riscv64-uefi-msvc: error: the following command failed with 1 compilation errors:
C:\Users\User\AppData\Local\Microsoft\WinGet\Packages\zig.zig.nightly_Microsoft.Winget.Source_8wekyb3d8bbwe\zig-windows-x86_64-0.12.0-dev.3+9c05810be\zig.exe build-exe D:\Azzam\kora\headstart\src\main.zig -cflags -ffreestanding -nostdlib -mno-red-zone -- D:\Azzam\kora\headstart\external\flanterm\flanterm.c D:\Azzam\kora\headstart\external\flanterm\backends\fb.c --cache-dir D:\Azzam\kora\headstart\zig-cache --global-cache-dir C:\Users\User\AppData\Local\zig --name headstart -mcmodel medium -target riscv64-uefi-msvc -mcpu baseline_rv64 --mod build_options::D:\Azzam\kora\headstart\zig-cache\c\617d0a7ef579f0c5c8f720583a81a2fe\options.zig --deps build_options -I D:\Azzam\kora\headstart\external --listen=- 
Build Summary: 1/6 steps succeeded; 1 failed (disable with --summary none)
run transitive failure
└─ run sh transitive failure
   └─ install transitive failure
      └─ install headstart transitive failure
         └─ zig build-exe headstart Debug riscv64-uefi-msvc 1 errors
D:\Azzam\kora\headstart\zig-cache\o\ba881674b31c5b5c615cc912c8122d92\cimport.zig:207:175: error: @bitCast size mismatch: destination type 'usize' has 64 bits but source type 'c_long' has 32 bits
    return flanterm_fb_init(null, null, framebuffer, width, height, pitch, null, null, null, null, null, null, null, @as(?*anyopaque, @ptrFromInt(@as(c_int, 0))), @as(usize, @bitCast(@as(c_long, @as(c_int, 0)))), @as(usize, @bitCast(@as(c_long, @as(c_int, 0)))), @as(usize, @bitCast(@as(c_long, @as(c_int, 1)))), @as(usize, @bitCast(@as(c_long, @as(c_int, 1)))), @as(usize, @bitCast(@as(c_long, @as(c_int, 1)))), @as(usize, @bitCast(@as(c_long, @as(c_int, 0)))));
                                                                                                                                
                                              ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
referenced by:
    init: src\console\flanterm.zig:11:16
    main: src\main.zig:25:12
    remaining reference traces hidden; use '-freference-trace' to see all reference traces
```

### Expected Behavior

It should compiles fine like in x86_64 and aarch64.

### LINKED ISSUE #18543: `translate-c`: negative pointer to void
Take this C for example:
```c
#include <stdio.h>

int main() {
    void *x = (void *)-1;
    printf("ptr: %p\n", x);
}
```

when compiled with `gcc` the result is:
```
ptr: 0xffffffffffffffff
```

however, when compiler with `translate-c`
```
out.zig:295:55: error: type 'usize' cannot represent integer value '-1'
    var x: ?*anyopaque = @as(?*anyopaque, @ptrFromInt(-@as(c_int, 1)));
```

the solution would be to use `-%` wrapping negation for the `-1` I guess.

### PR DESCRIPTION
**Pull request by f-cozzocrea**
Fixes an issue where translate-c would not create the correct cast between pointers and non-usize types (or int types that don't coerce to usize). This fix generates correct code that casts between pointers and C integer types of any width or signedness. 

My previous PR produced non-portable code. This PR includes a check in the generated code for the sizes of the int and pointer types.

Closes #18543, closes #18178, closes #17427.

Supersedes #18595.

### CODE REVIEW

File: src/translate_c.zig [hunk 1]
```diff
@@ -2357,20 +2357,176 @@ fn transCCast(
...
+        //
+        // Example:
+        //  (blk: {
+        //      if(@sizeOf(src_ptr_type) > @sizeOf(dst_int_type)) {
+        //          break :blk @as(c_ulong, @truncate(@intFromPtr(expr)));
+        //      }
+        //      else {
+        //          break :blk @as(c_ulong, @intFromPtr(expr));
+        //      }
+        //      break blk: _tmp
```
Comment (ehaas) [inline src/translate_c.zig:2379]: This `break blk: _tmp` line seems like it was inadvertently included in the comment?

Comment (f-cozzocrea) [reply src/translate_c.zig:2379]: Good catch, I'll remove that.

**Comment by f-cozzocrea** [comment]
After thinking about it more, I found out that zig supports vector casting now, so the path forward seems clear.

**Comment by f-cozzocrea** [comment]
Closing this for now. I'll probably come back to it in the future, or add it to the aro implementation instead.

**Pull request closed without merging**